firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/Personalization.java

```diff
@@ -17,18 +17,36 @@
 import android.os.Bundle;
 import androidx.annotation.NonNull;
 import com.google.firebase.analytics.connector.AnalyticsConnector;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 import org.json.JSONObject;
 
 public class Personalization {
   public static final String ANALYTICS_ORIGIN_PERSONALIZATION = "fp";
-  public static final String ANALYTICS_PULL_EVENT = "_fpc";
-  public static final String ARM_KEY = "_fpid";
-  public static final String ARM_VALUE = "_fpct";
-  static final String PERSONALIZATION_ID = "personalizationId";
+
+  // Constants with LOG_KEY suffix are how the corresponding ones without it are identified on
+  // Google Analytics.
+  public static final String ANALYTICS_PULL_EVENT = "personalization_assignment";
+  public static final String ARM_KEY = "arm_key";
+  public static final String ARM_VALUE = "arm_value";
+  public static final String PERSONALIZATION_ID = "personalizationId";
+  public static final String PERSONALIZATION_ID_LOG_KEY = "personalization_id";
+  public static final String ARM_INDEX = "armIndex";
+  public static final String ARM_INDEX_LOG_KEY = "arm_index";
+  public static final String GROUP = "group";
+
+  public static final String ANALYTICS_PULL_EVENT_INTERNAL = "_fpc";
+  public static final String CHOICE_ID = "choiceId";
+  public static final String CHOICE_ID_LOG_KEY = "_fpid";
 
   /** The app's Firebase Analytics client. */
   private final AnalyticsConnector analyticsConnector;
 
+  /** A map of Remote Config parameter key to choice ID. */
+  private final Map<String, String> armsCache =
```

Comment:
I'd still like to rename this to something more descriptive. Maybe `loggedArms` or `loggedChoiceIds`?

Reply:
👍  I agree that'd be helpful